x-pack/plugins/security/public/management/roles/edit_role/privileges/kibana/space_aware_privilege_section/space_aware_privilege_section.tsx

```diff
@@ -218,7 +218,7 @@ export class SpaceAwarePrivilegeSection extends Component<Props, State> {
     const viewMatrixButton = (
       <PrivilegeSummary
         role={this.props.role}
-        spaces={this.getDisplaySpaces()}
+        spaces={this.getSelectedSpaces()}
```

Comment:
>[!CAUTION]
> There is something definitely off here when I select different access variations. See the video. Do you see the same?

https://github.com/user-attachments/assets/282cafa0-7ae6-46bc-9699-3edb7f24dd6c

Reply:
Confirmed, can reproduce, will take a look

Reply:
6c631b41973d0ed5bba6a9a2ebe86b1ed7d66abd pushed fix, but please hold on review for now, need to check tests

Reply:
@azasypkin ready for review 🙂

Reply:
Nice! I'll review on Monday

Reply:
It looks like we have another issue now. If Read/All access to Fleet is granted through a Kibana privilege entry with “All Spaces”, the privilege summary for any _space-specific_ privileges should reflect what is granted by the “All Spaces” privilege.


https://github.com/user-attachments/assets/a9cfcd14-b31b-40ec-9cfb-882c1d4503f8